### PR TITLE
wsd: Unconditionally compile support-key-enabled code paths

### PR DESCRIPTION
**Pull request by thebearon**
Change-Id: Iec477c223f2dc75e0be8b472c8439ca9311d6aae

* Target version: master 

### Checklist

- [X] Code is properly formatted
- [X] All commits have Change-Id
- [ ] I have run tests with `make check`
- [X] I have issued `make run` and manually verified that everything looks okay
- [X] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: wsd/COOLWSD.cpp [hunk 1]
```diff
@@ -2681,44 +2682,45 @@ void COOLWSD::innerInitialize(Application& self)
...
-    {
-        LOG_WRN("Support key not set, please use 'coolconfig set-support-key'.");
-        std::cerr << "Support key not set, please use 'coolconfig set-support-key'." << std::endl;
-        COOLWSD::OverrideWatermark = "Unsupported, the support key is missing.";
-    }
-    else
+    if (config::isSupportKeyEnabled())
     {
-        SupportKey key(supportKeyString);
+        const std::string supportKeyString = getConfigValue<std::string>(conf, "support_key", "");
```
Comment (Ashod) [inline wsd/COOLWSD.cpp:2687]: Should be `static`. Also, `""` is unidiomatic, use `std::string()`.

File: wsd/FileServer.cpp [hunk 2]
```diff
@@ -1212,15 +1210,16 @@ FileServerRequestHandler::ResourceAccessDetails FileServerRequestHandler::prepro
...
 
-#if ENABLE_SUPPORT_KEY
-    const std::string keyString = config.getString("support_key", "");
-    SupportKey key(keyString);
-    if (!key.verify() || key.validDaysRemaining() <= 0)
+    if (config::isSupportKeyEnabled())
     {
-        brandCSS = Poco::format(linkCSS, responseRoot, std::string(BRANDING_UNSUPPORTED));
-        brandJS = Poco::format(scriptJS, responseRoot, std::string(BRANDING_UNSUPPORTED));
+        const std::string keyString = config.getString("support_key", "");
```
Comment (Ashod) [inline wsd/FileServer.cpp:1215]: Same as the other case: `static` and `std::string()`.

File: wsd/FileServer.cpp [hunk 3]
```diff
@@ -1594,17 +1593,18 @@ void FileServerRequestHandler::preprocessAdminFile(const HTTPRequest& request,
...
-    const std::string keyString = config.getString("support_key", "");
-    SupportKey key(keyString);
-
-    if (!key.verify() || key.validDaysRemaining() <= 0)
+    if (config::isSupportKeyEnabled())
     {
-        brandJS = Poco::format(scriptJS, std::string(BRANDING_UNSUPPORTED));
-        brandFooter = Poco::format(footerPage, key.data(), Poco::DateTimeFormatter::format(key.expiry(), Poco::DateTimeFormat::RFC822_FORMAT));
+        const auto& config = Application::instance().config();
+        const std::string keyString = config.getString("support_key", "");
```
Comment (Ashod) [inline wsd/FileServer.cpp:1599]: Same.

**Review by Ashod — APPROVED** [review]
Thanks, @thebearon. Left some notes and suggestions.

(This will cause some conflicts in FileServer for me.)

**Comment by thebearon** [comment]
Thanks for the review, @Ashod! Let me think about and address those comments in a follow-up PR once this is in.
Let's see if it compiles on iOS now.

**Review by Ashod — APPROVED** [review]
Thanks, @thebearon. Good stuff.

**Pull request merged**